sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -123,13 +124,23 @@ public Long addSample(final Sample sample) throws IOException {
   }
 
   @Override
-  public int getNextSiblingNumber(Sample parent, SampleClass childClass) throws IOException {
-    Query query = currentSession().createQuery(
-        "select max(siblingNumber) " + "from DetailedSampleImpl " + "where parentId = :parentId " + "and sampleClassId = :sampleClassId");
-    query.setLong("parentId", parent.getId());
-    query.setLong("sampleClassId", childClass.getId());
+  public int getNextSiblingNumber(String partialAlias) throws IOException {
+    Query query = currentSession().createQuery("select max(siblingNumber) from DetailedSampleImpl as ds"
+            + " where alias IN (concat(:alias, ds.siblingNumber), concat(:alias, '0', ds.siblingNumber))");
+    query.setString("alias", partialAlias);
     Number result = ((Number) query.uniqueResult());
-    return result == null ? 1 : result.intValue() + 1;
+    int next = result == null ? 0 : result.intValue();
+
+    // verify uniqueness and fix siblingNumbers for existing samples if necessary
```

Comment:
Updated javadoc/comments to clarify a bit. Did not explain scope because scope is not a real thing here, and there is nothing to suggest that it is. With our naming scheme, there is an imagined/effective scope, which is different depending on class and category, but sibling number is really just the lowest possible number that forms a unique alias. With another naming scheme, the imagined/effective scope could be completely different.